src/particles/wakefields/HandleWakefield.H

```diff
@@ -0,0 +1,150 @@
+            amrex::Gpu::DeviceVector<amrex::Real> mean_x(num_bins, 0.0);
+            amrex::Gpu::DeviceVector<amrex::Real> mean_y(num_bins, 0.0);
+
+            // Call charge deposition function
+            impactx::particles::wakefields::DepositCharge1D(particle_container, charge_distribution, bin_min, bin_size, is_unity_particle_weight);
+
+            // Sum up all partial charge histograms to one MPI process to calculate the global wakefield.
+            // Once calculated, we will distribute convoluted_wakefield back to every MPI process.
+            amrex::ParallelReduce::Sum(
+                charge_distribution.data(),
+                charge_distribution.size(),
+                amrex::ParallelDescriptor::IOProcessorNumber(),
+                amrex::ParallelDescriptor::Communicator()
+            );
+
+            amrex::Gpu::DeviceVector<amrex::Real> convoluted_wakefield;
+            if (amrex::ParallelDescriptor::IOProcessor()) {
+                // Call the mean transverse position function
+                impactx::particles::wakefields::MeanTransversePosition(particle_container, mean_x, mean_y, bin_min,
+                                                                       bin_size, is_unity_particle_weight);
+
+                // Call charge density derivative function
+                amrex::Gpu::DeviceVector<amrex::Real> slopes(charge_distribution.size() - 1, 0.0);
+                impactx::particles::wakefields::DerivativeCharge1D(charge_distribution, slopes, bin_size,
+                                                                   GetNumberDensity); // Use number derivatives for convolution with CSR
+
+                // Construct CSR wake function on 2N support
+                amrex::Gpu::DeviceVector<amrex::Real> wake_function(num_bins*2, 0.0);
+                amrex::Real *const dptr_wake_function = wake_function.data();
+                auto const dR = R;  // for NVCC capture
+                amrex::ParallelFor(num_bins*2, [=] AMREX_GPU_DEVICE(int i) {
+                    if (i < num_bins) {
+                        amrex::Real const s = static_cast<amrex::Real>(i) * bin_size;
+                        dptr_wake_function[i] = impactx::particles::wakefields::w_l_csr(s, dR, bin_size);
+                    }
+                    else if (i > num_bins) {
+                        amrex::Real const s = static_cast<amrex::Real>(i - 2*num_bins) * bin_size;
+                        dptr_wake_function[i] = impactx::particles::wakefields::w_l_csr(s, dR, bin_size);
+                    }
+                });
+
+                // Call convolution function
+                convoluted_wakefield = impactx::particles::wakefields::convolve_fft(slopes, wake_function, bin_size);
+            }
+
+            // Broadcast the global wakefield to every MPI rank
+            amrex::ParallelDescriptor::Bcast(
+                convoluted_wakefield.data(),
+                convoluted_wakefield.size(),
```

Comment:
```suggestion
                convolved_wakefield.size(),
```

Reply:
I originally thought we should change "convoluted" -> "convolved" for grammatical reasons, but it is used in many places.  I think it's ok to keep as-is.